compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -504,12 +558,14 @@ impl<'tcx> Stable<'tcx> for ty::GenericArgs<'tcx> {
     }
 }
 
-impl<'tcx> Stable<'tcx> for ty::PolyFnSig<'tcx> {
-    type T = stable_mir::ty::PolyFnSig;
-    fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
-        use stable_mir::ty::Binder;
+impl<'tcx, S, T> Stable<'tcx> for ty::Binder<'tcx, S>
+where
+    S: Stable<'tcx, T = T> + Copy,
```

Comment:
Unsure what's the preferred way here, if having Copy as bound, if using `Deref` and calling as_deref() and then implement stable for what's returned or if calling clone.
cc @oli-obk

Reply:
I considered using a `Clone` bound but I thought it might be too weak?

If there's a type with an expensive clone, I'm not sure I wanted to implicitly derive an expensive implementation of stable

Reply:
Also, isn't `as_deref` just an `Option` thing? I thought the method you get from `Deref` is just `deref`

Reply:
If I'm not wrong, this works ...

```diff
diff --git a/compiler/rustc_smir/src/rustc_smir/mod.rs b/compiler/rustc_smir/src/rustc_smir/mod.rs
index dd0d7df9ecf..b7e827dced0 100644
--- a/compiler/rustc_smir/src/rustc_smir/mod.rs
+++ b/compiler/rustc_smir/src/rustc_smir/mod.rs
@@ -581,13 +581,13 @@ fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {

 impl<'tcx, S, T> Stable<'tcx> for ty::Binder<'tcx, S>
 where
-    S: Stable<'tcx, T = T> + Copy,
+    S: Stable<'tcx, T = T>,
 {
     type T = stable_mir::ty::Binder<T>;

     fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
         stable_mir::ty::Binder {
-            value: self.skip_binder().stable(tables),
+            value: self.as_ref().skip_binder().stable(tables),
             bound_vars: self
                 .bound_vars()
                 .iter()
```

Reply:
Oh `as_ref` not `as_deref`

Reply:
Oh ignore me I'm being dumb--binder has both